File: Difficulty: Easy/Missing in Array/missing-in-array.py
# User function Template for python3
class Solution:
    def missingNum(self, arr):
        n = len(arr) + 1  # The total numbers should be from 1 to n
        expected_sum = n * (n + 1) // 2
        actual_sum = sum(arr)
        return expected_sum - actual_sum

File: Difficulty: Easy/Missing in Array/missing-in-array.cpp
class Solution {
  public:
    int missingNum(vector<int>& arr) {
        int n = arr.size() + 1;  // original size should be n
        long long total = (1LL * n * (n + 1)) / 2; // sum of 1..n
        long long sum = 0;
        
        for (int num : arr) {
            sum += num;
        }
        
        return total - sum; // missing number
    }
};